### PR TITLE
Fix #372 updates statedumpable implementation

### LINKED ISSUE #372: update statedumpable implementation to match new api changes
The StateDumpable interface has changed.  The method addState(String, String) has changed to addState(String, Object).  There were many workarounds to pass the 2nd parameter as a String.  These changes have removed those workarounds.

### PR DESCRIPTION
**Pull request by gibsong**
updates statedumpable implementation to match the new api, 2nd arg is now Object not String.

### CODE REVIEW

File: management/testing/sample-entity/src/main/java/org/terracotta/management/entity/sample/server/MapProvider.java [hunk 1]
```diff
@@ -91,10 +91,6 @@ public void close() {
 
   @Override
   public void addStateTo(StateDumpCollector stateDumper) {
-    StateDumpCollector caches = stateDumper.subStateDumpCollector("caches");
-    int c = 0;
-    for (String name : this.caches.keySet()) {
-      caches.addState(String.valueOf(c), name);
```
Comment (mathieucarbou) [inline management/testing/sample-entity/src/main/java/org/terracotta/management/entity/sample/server/MapProvider.java]: damn! there was a bug lol! hopefully we changed ;-)

File: voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java [hunk 2]
```diff
@@ -108,11 +110,8 @@ public final void addStateTo(StateDumpCollector stateDumpCollector) {
     // clients, by default for all active entities
     Collection<ClientDescriptor> clients = getClients();
     stateDumpCollector.addState("clientCount", String.valueOf(clients.size()));
-    StateDumpCollector clientsDumper = stateDumpCollector.subStateDumpCollector("clients");
-    int i = 0;
-    for (ClientDescriptor client : clients) {
-      clientsDumper.addState(String.valueOf(i++), client.toString());
-    }
+    stateDumpCollector.addState("clients",clients);
```
Comment (mathieucarbou) [inline voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java]: @gibsong : clients is a list of ClientDescriptor: did you verified in the state dump output that it works ? Or we have to convert it to a list of string before ?

File: voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java [hunk 3]
```diff
@@ -15,6 +15,7 @@
  */
 package org.terracotta.voltron.proxy.server;
 
+import com.sun.java.swing.plaf.windows.WindowsTreeUI;
```
Comment (myronkscott) [inline voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java]: ??

File: management/testing/sample-entity/src/main/java/org/terracotta/management/entity/sample/server/MapProvider.java (hunk 1 above)
Comment (gibsong) [inline management/testing/sample-entity/src/main/java/org/terracotta/management/entity/sample/server/MapProvider.java]: yeah now it is working better, was only including one cache and there were two.

File: voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java (hunk 3 above)
Comment (gibsong) [inline voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java]: hmm... dang IDE.  I will remove that import.  Not sure how that happened.

File: voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java (hunk 2 above)
Comment (gibsong) [inline voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java]: let me double check.  I thought it was outputting correctly

Comment (gibsong) [inline voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java]: Here are both outputs, clientsXXX show my current changes and clientsYYY shows previous state:
entityState={
        instance=org.terracotta.management.entity.nms.agent.server.ActiveNmsAgentServerEntity@196a5d1a
        clientCount=2
        clientsXXX=[
          ClientDescriptorImpl{nodeID=ClientID[1], entityDescriptor=ClientInstanceID(1)}
          ClientDescriptorImpl{nodeID=ClientID[2], entityDescriptor=ClientInstanceID(1)}
        ]
        clientsYYY={
          0=ClientDescriptorImpl{nodeID=ClientID[1], entityDescriptor=ClientInstanceID(1)}
          1=ClientDescriptorImpl{nodeID=ClientID[2], entityDescriptor=ClientInstanceID(1)}
        }
      }

**Comment by gibsong** [comment]
i renamed to commit to start with Close #372

File: voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java (hunk 2 above)
Comment (mathieucarbou) [inline voltron-proxy/voltron-proxy-server/src/main/java/org/terracotta/voltron/proxy/server/ActiveProxiedServerEntity.java]: yes!

**Pull request merged**